pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -605,6 +610,41 @@ func (b *BridgePodInterface) createBridge() error {
 	return nil
 }
 
+func (b *BridgePodInterface) switchPodInterfaceWithDummy() error {
+	originalPodInterfaceName := b.podInterfaceName
+	newPodInterfaceName := fmt.Sprintf("%s-nic", originalPodInterfaceName)
+	dummy := &netlink.Dummy{LinkAttrs: netlink.LinkAttrs{Name: originalPodInterfaceName}}
+
+	currentLink, _ := Handler.LinkByName(originalPodInterfaceName)
+	if currentLink.Type() != "dummy" {
+		// Rename pod interface to free the original name for a new dummy interface
+		err := Handler.LinkSetName(b.podNicLink, newPodInterfaceName)
+		if err != nil {
+			log.Log.Reason(err).Errorf("failed to rename interface : %s", b.podInterfaceName)
+			return err
+		}
+		b.podInterfaceName = newPodInterfaceName
+
+		// Create a dummy interface named after the original interface
+		err = Handler.LinkAdd(dummy)
+		if err != nil {
+			log.Log.Reason(err).Errorf("failed to create dummy interface : %s", newPodInterfaceName)
+			return err
+		}
+	}
+
+	// Replace original pod interface IP address to the dummy
+	// Since the dummy is not connected to anything, it should not affect networking
+	// Replace will add if ip doesn't exist or modify the ip
```

Comment:
The CNI checks that the interface exists with the ip it expects.

Reply:
> The CNI checks that the interface exists with the ip it expects.

I am pretty sure that the kubelet itself does this, and not CNI, on restart to detect if it may have been interrupted in a bad moment.